pinot-query-runtime/src/main/java/org/apache/pinot/query/service/QueryServer.java

```diff
@@ -45,13 +48,16 @@ public class QueryServer extends PinotQueryWorkerGrpc.PinotQueryWorkerImplBase {
 
   private final Server _server;
   private final QueryRunner _queryRunner;
+  private final ExecutorService _executorService;
 
   public QueryServer(int port, QueryRunner queryRunner) {
     _server = ServerBuilder.forPort(port).addService(this).build();
     _queryRunner = queryRunner;
+    _executorService = Executors.newFixedThreadPool(ResourceManager.DEFAULT_QUERY_RUNNER_THREADS,
```

Comment:
can we avoid creating another pool and instead reuse the pools that already exist? maybe we should pass this one into the query runner and then into the OpChainSchedulerService, or the opposite

Reply:
I am a bit confused with this. Is this new pool just to handle async submission to dispatcher ?

Reply:
yes I can reuse the worker pool. the reason why I did this is b/c there's a RUNNER thread and WORKER thread in ResourceManager for V1. 
- RUNNER is used for planMaker and running the instance and combine operator
- WORKER is used for executing individual tasks within combine operator 

I was trying to do the same. but I can definitely reuse either the RUNNER from V1 or WORKER from V2. i just don't know what's the best abstraction so i choose to follow what V1 has

Reply:
that's a good point! actually i double check the code it should reuse the same threadpool but allocate number of threads in an additive fashion. will reuse the same threadpool here